lib/base/application.cpp

```diff
@@ -885,6 +908,15 @@ void Application::ExceptionHandler()
 #ifdef _WIN32
 LONG CALLBACK Application::SEHUnhandledExceptionFilter(PEXCEPTION_POINTERS exi)
 {
+	/* If an unhandled C++ exception occurs with both a termination handler (std::set_terminate()) and an unhandled
+	 * SEH filter (SetUnhandledExceptionFilter()) set, the latter one is called, however our termination handler is
+	 * better suited for dealing with C++ exceptions. In this case, the SEH exception will have a specific code and
+	 * we can just call the default filter function which will take care of calling the termination handler.
+	 */
+	if (exi->ExceptionRecord->ExceptionCode == EXCEPTION_CODE_CXX_EXCEPTION) {
```

Comment:
Btw why did you outsource EXCEPTION_CODE_CXX_EXCEPTION at all?

Reply:
I think it's a good idea to name magic constants. Unfortunately I could not find a constant in a system header.